### PR TITLE
staking L2

### PR DESCRIPTION
**Pull request by clementbalestrat**

### CODE REVIEW

File: containers/Connector/Connector.tsx [hunk 1]
```diff
@@ -35,21 +36,22 @@ const useConnector = () => {
...
+			const provider = loadProvider({
 				networkId,
-				process.env.NEXT_PUBLIC_INFURA_PROJECT_ID
-			);
+				infuraId: process.env.NEXT_PUBLIC_INFURA_PROJECT_ID,
+				provider: window.ethereum,
+			});
 
+			// Hack until Synthetix lib can handle OVM chain ID
+			const useOvm = networkId === 10;
```
Comment (vbstreetz) [inline containers/Connector/Connector.tsx:47]: would be cool to also add kovan support? id = 69

**Review by vbstreetz — APPROVED** [review]

**Review by JChiaramonte7 — COMMENTED** [review]
Should we remove notify from the readme and .env.local example?

Also not 100% sure where this is happening, but on L2 the debtBalance/stakedValue don't update right away. Maybe forcing a react-query cache invalidation onTxConfirm could fix and make dapp feel more responsive

File: components/GasSelector/GasSelector.tsx [hunk 2]
```diff
@@ -35,36 +36,42 @@ const GasSelector: React.FC<GasSelectorProps> = ({
...
 	const [customGasPrice, setCustomGasPrice] = useRecoilState(customGasPriceState);
+	const network = useRecoilValue(networkState);
 
 	const exchangeRatesQuery = useExchangeRatesQuery();
 	const { selectedPriceCurrency } = useSelectedPriceCurrency();
 	const ethGasStationQuery = useEthGasStationQuery();
 
 	const gasPrices = ethGasStationQuery.data ?? ({} as GasPrices);
 	const exchangeRates = exchangeRatesQuery.data ?? null;
+	const isL2 = network?.useOvm ?? false;
```
[the remark this answers is not in the document]
Comment (dvd-schwrtz) [reply components/GasSelector/GasSelector.tsx:47]: it is also good to add to recoil if we are going to use the new data library which takes the isL2 boolean as an argument when creating an instance

[the remark this answers is not in the document]
Comment (clementbalestrat) [reply components/GasSelector/GasSelector.tsx:47]: Yeah that's a good point. We are using it everywhere. I'll make the change.

File: package.json [hunk 3]
```diff
@@ -16,16 +16,17 @@
...
 	"dependencies": {
 		"@artsy/fresnel": "1.3.0",
-		"@eth-optimism/watcher": "0.0.1-alpha.8",
 		"@metamask/detect-provider": "1.2.0",
 		"@reach/dialog": "0.12.1",
 		"@snapshot-labs/snapshot.js": "0.1.6",
 		"@synthetixio/assets": "2.0.7",
-		"@synthetixio/js": "2.39.3",
+		"@synthetixio/contracts-interface": "^0.0.7",
+		"@synthetixio/optimism-networks": "^0.0.10",
```
Comment (evgenyboxer) [inline package.json:24]:
```suggestion
		"@synthetixio/optimism-networks": "0.0.10",
```

File: package.json [hunk 4]
```diff
@@ -16,16 +16,17 @@
...
+		"@synthetixio/providers": "^0.0.11",
```
Comment (evgenyboxer) [inline package.json:25]:
```suggestion
		"@synthetixio/providers": "0.0.11",
```

File: package.json [hunk 5]
```diff
@@ -16,16 +16,17 @@
...
+		"@synthetixio/transaction-notifier": "^0.0.8",
```
Comment (evgenyboxer) [inline package.json:26]:
```suggestion
		"@synthetixio/transaction-notifier": "0.0.8",
```

File: package.json [hunk 6]
```diff
@@ -49,6 +50,7 @@
 		"react-select": "3.1.0",
 		"react-slick": "0.27.13",
 		"react-table": "7.5.0",
+		"react-toastify": "^7.0.3",
```
Comment (evgenyboxer) [inline package.json:53]:
```suggestion
		"react-toastify": "7.0.3",
```

File: sections/layer2/index.tsx [hunk 7]
```diff
@@ -1,35 +1,36 @@
...
-	const escrowDataQuery = useEscrowDataQuery();
-	const totalBalancePendingMigration = escrowDataQuery?.data?.totalBalancePendingMigration ?? 0;
+	const network = useRecoilValue(networkState);
 
 	useEffect(() => {
 		async function getData() {
 			try {
-				const provider = new providers.StaticJsonRpcProvider(OVM_RPC_URL);
+				const provider = getOptimismProvider({
+					layerOneNetworkId: network?.id ?? 1,
```
Comment (evgenyboxer) [inline sections/layer2/index.tsx:32]: perhaps this could default to some `MAINNET` constant?

**Review by evgenyboxer — APPROVED** [review]
@clementbalestrat awesome work! I left some code suggestions.

**Comment by clementbalestrat** [comment]
> Should we remove notify from the readme and .env.local example?

Yep good idea. Just fixed that.
 
> Also not 100% sure where this is happening, but on L2 the debtBalance/stakedValue don't update right away. Maybe forcing a react-query cache invalidation onTxConfirm could fix and make dapp feel more responsive

Yeah I think we need to refetch pretty much every where where a tx gets confirmed. Can be done in another PR though.

**Pull request merged**